[TensorPipe] Use Descriptor::Tensor::sourceDevice in tensorpipe_agent.

beauby opened the pull request:
Test Plan: CI

Reviewed By: lw

Differential Revision: D27661608

mrshenli commented on torch/csrc/distributed/rpc/tensorpipe_utils.h:
```diff
@@ -14,6 +14,13 @@
+#else
+using Allocation = Message;
+using Descriptor = Message;
+#endif // TP_NEW_API
```
I assume there will be an immediate PR to remove these aliases?

beauby replied:
Right, this preprocessor switch is introduced to allow a clean update in fbcode and GitHub at the same time. The `TP_NEW_API` switch will never actually be enabled in GitHub, and the `#ifdef` will simply disappear in the next PR.

mrshenli reviewed (approved):
LGTM! I don't think there will be any conflicts with https://github.com/pytorch/pytorch/pull/54932, but it might worth checking.

Commit:
[TensorPipe] Prepare for new Pipe API.

Test Plan: CI

Differential Revision: D27648291

fbshipit-source-id: ca34b34ccbd1695bc117381b22d9d2cc4651cecc

Commit:
[TensorPipe] Use Descriptor::Tensor::sourceDevice in tensorpipe_agent. (#55821)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/55821

Test Plan: CI

Reviewed By: lw

Differential Revision: D27661608

fbshipit-source-id: eea96045cb1fb695012821a5a127f52c0a6d46b7

mrshenli commented:
rocm test failures look irrelevant. 

```
18:14:37   test_Conv1d (__main__.TestNN) ... Traceback (most recent call last):
18:14:38   File "test/run_test.py", line 1101, in <module>
18:14:38     main()
18:14:38   File "test/run_test.py", line 1080, in main
18:14:38     raise RuntimeError(err_message)
18:14:38 RuntimeError: test_nn failed! Received signal: SIGSEGV
```

```
20:19:43   test_autocast_torch_fp16 (__main__.TestCuda) ... Traceback (most recent call last):
20:19:55   File "test/run_test.py", line 1101, in <module>
20:19:55     main()
20:19:55   File "test/run_test.py", line 1080, in main
20:19:55     raise RuntimeError(err_message)
20:19:55 RuntimeError: test_cuda failed! Received signal: SIGSEGV
```
